rollup.config.js

```diff
@@ -190,45 +194,66 @@ export function makeBaseBundleConfig(options) {
   return deepMerge(sharedBundleConfig, isAddOn ? addOnBundleConfig : standAloneBundleConfig);
 }
 
-export function makeMinificationVariants(existingConfigs) {
-  const newConfigs = [];
-
-  // ensure we've got an array of configs rather than a single config
-  existingConfigs = Array.isArray(existingConfigs) ? existingConfigs : [existingConfigs];
+/**
+ * Takes the CDN rollup config for a given package and produces three versions of it:
+ *   - non-minified, including debug logging,
+ *   - minified, including debug logging,
+ *   - minified, with debug logging stripped
+ *
+ * @param baseConfig The rollup config shared by the entire package
+ * @returns An array of versions of that config
+ */
+export function makeConfigVariants(baseConfig) {
+  const configVariants = [];
 
-  existingConfigs.forEach(existingConfig => {
-    const { plugins } = existingConfig;
+  const { plugins } = baseConfig;
+  const includeDebuggingPlugin = makeIsDebugBuildPlugin(true);
+  const noDebuggingPlugin = makeIsDebugBuildPlugin(false);
 
-    // The license plugin has to be last, so it ends up after terser. Otherwise, terser will remove the license banner.
-    assert(
-      getLastElement(plugins).name === 'rollup-plugin-license',
-      `Last plugin in given options should be \`rollup-plugin-license\`. Found ${getLastElement(plugins).name}`,
-    );
+  // The license plugin has to be last, so it ends up after terser. Otherwise, terser will remove the license banner.
+  assert(
+    getLastElement(plugins).name === 'rollup-plugin-license',
+    `Last plugin in given options should be \`rollup-plugin-license\`. Found ${getLastElement(plugins).name}`,
+  );
 
-    const bundleVariants = [
-      {
-        output: {
-          file: `${existingConfig.output.file}.js`,
-        },
-        plugins,
+  // The additional options to use for each variant we're going to create #namingishard
+  const variantSpecificOptionsVariants = [
```

Comment:
I might be off here (and agreed, #namingissuperhard) but based on the comment, why don't we go with `variantSpecificOptions` here? or perhaps something like `bundleSpecificVariants`? I don't know if these are particularly better, so feel free to ignore this comment 😅

Reply:
Ugh. I struggled so hard with this one. My basic problem is that it's a double plural (each individual one is a set of options, and you have options for which set of options you want 😓). `variantSpecificOptions` could easily be a single set of options specific to a particular variant. I wish language had parentheses the way math does. Then I'd just call it `(variantSpecificOptions)s` and be done with it.

I don't love it (as far as I'm concerned, "configuration" is a mass noun, not a count noun, and therefore shouldn't be pluralized), but what about `variantConfigs`?

Reply:
Sure that sounds good as well (also short and simple which is probably good). 
As I said though, if you prefer the original naming, that's fine with me as well. It was just something that I tried to come up with a better name but as we've seen, it seems like there is no perfect solution in this case 😅

Reply:
No, you weren’t wrong. I wouldn't have felt the need to make a joke about how terrible the name was otherwise!

I'll go with `variantConfigs`.